powerscale/models/snapshot.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright (c) 2023 Dell Inc., or its subsidiaries. All Rights Reserved.
+
+Licensed under the Mozilla Public License Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://mozilla.org/MPL/2.0/
+
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package models
+
+import "github.com/hashicorp/terraform-plugin-framework/types"
+
+// SnapshotDataSourceModel describes the data source data model.
+type SnapshotDataSourceModel struct {
+	ID        types.String          `tfsdk:"id"`
+	Snapshots []SnapshotDetailModel `tfsdk:"snapshots_details"`
+	// filter
+	SnapshotFilter *SnapshotFilterType `tfsdk:"filter"`
+}
+
+// SnapshotFilterType describes the filter data model.
+type SnapshotFilterType struct {
+	Path types.String `tfsdk:"path"`
```

Comment:
@doriac11 Can you add additional filters which are supported by the API itself like other datasources?

Reply:
@nikitajoshi1 , I read through those filters, but they didnt make much sense (to a user). I think for the most part users will want to filter on the snapshots related to a particular path.